lib/src/index_column.dart

```diff
@@ -28,6 +32,18 @@ class IndexColumn<T /* entity type */, I /* index type */ > {
         'Can not create index for field "$field", as type can not be asserted. Type is $I.',
       );
     }
+
+    if (referencedEntity != null &&
+        (_typeEqual<I, String?>() ||
+            _typeEqual<I, num?>() ||
+            _typeEqual<I, int?>() ||
+            _typeEqual<I, double?>() ||
+            _typeEqual<I, bool?>() ||
+            _typeEqual<I, DateTime?>())) {
+      throw Exception(
```

Comment:
Maybe we would like to `null` going forward as well, just like the underlying SQL implementation, in which case the foreign key link is ignored.